.github/actions/setup-build/action.yml

```diff
@@ -0,0 +1,357 @@
+        fi
+
+        if $feat_color || $feat_forcer; then
+          feat_formlib=true
+          items=''
+          $feat_color && items="$items-color"
+          $feat_forcer && items="$items-forcer"
+          echo "formlib-key=formlib$items" >>"$GITHUB_OUTPUT"
+          echo "formlib-restore-keys=formlib-" >>"$GITHUB_OUTPUT"
+          echo "FORMPATH=${{ github.workspace }}/formlib" >>"$GITHUB_ENV"
+        fi
+
+        # Output the feature variables.
+
+        for feat in $feature_names; do
+          value=$(eval "echo \$feat_${feat}")
+          echo "$feat=$value"
+          echo "$feat=$value" >>"$GITHUB_OUTPUT"
+        done
+
+        # Make CONFIG_ARGS.
+
+        args='--disable-dependency-tracking'
+
+        if $has_scalar; then
+          args="$args --enable-scalar"
+        else
+          args="$args --disable-scalar"
+        fi
+        if $has_threaded; then
+          args="$args --enable-threaded"
+        else
+          args="$args --disable-threaded"
+        fi
+        if $has_mpi; then
+          args="$args --enable-parform"
+        else
+          args="$args --disable-parform"
+        fi
+
+        if $feat_vorm || $feat_tvorm || $feat_parvorm; then
+          args="$args --enable-debug"
+        fi
+
+        if $feat_coverage; then
+          args="$args --enable-coverage"
+        fi
+
+        if $has_bin; then
+          if $feat_deploy; then
+            # Avoid compiler optimizations specific to the build host.
+            args="$args --disable-native"
+            # Enable static linking whenever possible.
+            if ${{ runner.os != 'macOS' }}; then
+              args="$args --enable-static-link"
+            fi
+          fi
+
+          if ${{ runner.os == 'Windows' }}; then
+            args="$args --with-api=windows"
+          else
+            args="$args --with-api=posix"
+          fi
+        else
+          # This avoids checking optimization flags.
+          args="$args --disable-native"
+        fi
+
+        if $feat_gmp; then
+          args="$args --with-gmp"
+        else
+          args="$args --without-gmp"
+        fi
+
+        if $feat_mpfr; then
+          args="$args --with-mpfr"
+        else
+          args="$args --without-mpfr"
+        fi
+
+        if $feat_zlib; then
+          args="$args --with-zlib"
+        else
+          args="$args --without-zlib"
+        fi
+
+        echo "CONFIG_ARGS=$args" >>"$GITHUB_ENV"
+
+        # Determine DISTNAME if possible.
+
+        if [ -f ./scripts/git-version-gen.sh ] && command -v git >/dev/null 2>&1; then
+          echo "DISTNAME=form-$(./scripts/git-version-gen.sh -r | sed '2q;d' | sed 's/^v//')" >>"$GITHUB_ENV"
+        fi
+
+    # awalsh128/cache-apt-pkgs-action does not allow empty packages argument.
+    # See: https://github.com/awalsh128/cache-apt-pkgs-action/issues/149
+    # As a workaround, we need to check if any packages need to be installed.
+    - name: Install dependencies (Ubuntu)
```

Comment:
This can be simplified if https://github.com/awalsh128/cache-apt-pkgs-action/pull/154 is merged.